Fix #290: Fix Custom Reports Entities for Moodle 4.5

jcorel opened the pull request:
This pull request updates the custom reports entities to use the get_default_tables() method instead of the deprecated get_default_table_aliases(), ensuring compatibility with Moodle 4.5.

Fixes issue #209 where generating custom reports triggered a deprecation warning in Moodle 4.5.

jcorel committed:
fix: update custom reports entities to use get_default_tables instead of deprecated get_default_table_aliases, in Moodle 4.5 (#209)

jcorel committed:
fix: update custom reports entities to use get_default_tables instead of deprecated get_default_table_aliases, in Moodle 4.5 (#209)

danmarsden commented:
Thanks for the PR! - I've just pushed through an updated branch specifcially for 4.5 and higher - it still ran the github actions on the older branches which are fine to ignore, but it's tripped the following "new" errors which we'll need to fix - if you could rebase your branch on the new 405_STABLE one and fix the below errors that would be great, otherwise I'll try to get to that soon myself sometime.

```
---

FOUND 1 ERROR AFFECTING 1 LINE

------------------------------------------------------------------------------------------------------------------------------------
 34 | ERROR | [x] Whitespace found at end of line (Squiz.WhiteSpace.SuperfluousWhitespace.EndLine)
------------------------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
------------------------------------------------------------------------------------------------------------------------------------

FILE: ...odle-local_recompletion/moodle-local_recompletion/moodle/local/recompletion/classes/reportbuilder/entities/hotpot_attempts.php
------------------------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
------------------------------------------------------------------------------------------------------------------------------------
 36 | ERROR | [x] Whitespace found at end of line (Squiz.WhiteSpace.SuperfluousWhitespace.EndLine)
------------------------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
------------------------------------------------------------------------------------------------------------------------------------

FILE: ...local_recompletion/moodle-local_recompletion/moodle/local/recompletion/classes/reportbuilder/entities/h5pactivity_attempts.php
------------------------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
------------------------------------------------------------------------------------------------------------------------------------
 36 | ERROR | [x] Whitespace found at end of line (Squiz.WhiteSpace.SuperfluousWhitespace.EndLine)
------------------------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
------------------------------------------------------------------------------------------------------------------------------------

jcorel committed:
reorder lang strings alphabetically

jcorel committed:
Merge branch 'fix/custom-reports-error-moodle-4-5' of https://github.com/jcorel/moodle-local_recompletion into fix/custom-reports-error-moodle-4-5

jcorel committed:
Small fixes to solve code checker errors.

jcorel commented:
Hi @danmarsden, thanks for the feedback! I've now rebased this branch onto `MOODLE_405_STABLE` and fixed the errors indicated by the code checker. The PR has been updated. Let me know if anything else is needed!
